ServiceSamples/TLSConsoleApplication/Program.cs

```diff
@@ -0,0 +1,80 @@
+
+namespace OAuthXppConsoleApplication
+{
+    class Program
+    {
+        public static string sessionUrl = "/api/services/UserSessionService/AifUserSessionService/GetUserSessionInfo";
+        
+        static void Main(string[] args)
+        {
+            string GetUserSessionOperationPath = string.Format("{0}{1}", ClientConfiguration.Default.UriString.TrimEnd('/'), sessionUrl);
+            
+            try
+            {
+                // Creates an HttpWebRequest for user session URL
+                HttpWebRequest aadRequest = (HttpWebRequest) WebRequest.Create(GetUserSessionOperationPath);
+
+                // Change TLS version of HTTP request if the TLS version value is defined in ClientConfiguration
+                if (!string.IsNullOrWhiteSpace(ClientConfiguration.OneBox.TLSVersion))
+                {
+                    aadRequest.ProtocolVersion = Version.Parse(ClientConfiguration.OneBox.TLSVersion);
+                }
+
+                string tlsRequestVersion = aadRequest.ProtocolVersion.ToString();                
+                Console.WriteLine("The TLS protocol version for the HTTP request is {0}.", tlsRequestVersion);
+                
+                aadRequest.Headers[OAuthHelper.OAuthHeader] = OAuthHelper.GetAuthenticationHeader();
+                aadRequest.Method = "POST";
+                aadRequest.ContentLength = 0;
+
+                // Get HttpWebResponse for the response
+                var aadResponse = (HttpWebResponse) aadRequest.GetResponse();
+
+                string tlsResponseVersion = aadResponse.ProtocolVersion.ToString();
+                Console.WriteLine("The TLS protocol version of the server response is {0}.", tlsResponseVersion);
+
+                if (aadResponse.StatusCode != HttpStatusCode.OK)
+                {
+                    Console.WriteLine("Could not get response from the server.");
+                    Console.ReadLine();
+                    return;
+                }
+
+                // Get response string
+                using (Stream responseStream = aadResponse.GetResponseStream())
+                {
+                    using (StreamReader streamReader = new StreamReader(responseStream))
+                    {
+                        string responseString = streamReader.ReadToEnd();
+
+                        Console.WriteLine(string.Format("Request sent using version {0}. Successfully received response with version {1}.", tlsRequestVersion, tlsResponseVersion));
+
+                        Console.WriteLine(string.Format("\nResponse string: {0}.", responseString));                        
+                    }
+                }
+                
+                // Releases the resources of the response.
+                aadResponse.Close();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Failed to authenticate to AAD with the exception: {0} and stack trace: {1}.", ex.ToString(), ex.StackTrace);
```

Comment:
This might not always be correct. The exception could for other reasons too.

Reply:
Authenticate to AAD is generic term. The code can fail when create the web request or when try to get response. Any of them is authentication error. What other reasons you think can happen here?

Reply:
For eg: once we get the response we are reading it. It could fail too. 
I was trying to say that the catch block is not for a specific exception but the message we are outputting is a specific one.

Reply:
Updated.